src/components/postgresConnect.ts

```diff
@@ -360,41 +360,27 @@ class PostgresDB {
     }
   }
 
-  public checkIfInFileTable = async (subPath: string) => {
+  public deleteMissingFileRecords = async (fileIds: number[]) => {
     const text = `
-    SELECT id 
-    FROM file
-    WHERE file_path = $1
-    `
-    try {
-      const result = await this.query({ text, values: [subPath] })
-      return result.rows.length === 0
-    } catch (err) {
-      throw err
-    }
-  }
-
-  public filePaths = async () => {
-    const text = `
-    SELECT file_path FROM file
+      DELETE FROM file
+      WHERE id = ANY($1);
     `
     try {
-      const result = await this.query({ text })
-      return result.rows
+      await this.query({ text, values: [fileIds] })
     } catch (err) {
       throw err
     }
   }
 
-  public setIsMissing = async (filePath: string) => {
+  public checkIfInFileTable = async (filePath: string) => {
     const text = `
-    UPDATE file
-    SET is_missing = true
-    WHERE file_path = $1
+    SELECT id 
+    FROM file
+    WHERE file_path = $1 OR thumbnail_path = $1
```

Comment:
This is a bit of a mess to read as it's mixing and matching two functions -- one new one existing. Check the actual file to make better sense of it. But there's not much different here -- "checkIfInFileTable" is basically the same, have replaced "setIsMissing" with "deleteMissingFileRecords"